mpp/shmemx.h4.in

```diff
@@ -55,6 +55,42 @@ include(shmemx_c_func.h4)dnl
 #ifdef __cplusplus
 } /* extern "C" */
 
+define(`SHMEM_CXX_WAIT_UNTIL_ALL',
+`static inline void shmemx_wait_until_all($2 *ivars, size_t nelems, int cmp, $2 cmp_value) {
+    shmemx_$1_wait_until_all(ivars, nelems, cmp, cmp_value);
+}')dnl
+SHMEM_CXX_DEFINE_FOR_SYNC(`SHMEM_CXX_WAIT_UNTIL_ALL')
+
+define(`SHMEM_CXX_WAIT_UNTIL_ANY',
+`static inline size_t shmemx_wait_until_any($2 *ivars, size_t nelems, int * restrict status, int cmp, $2 cmp_value) {
```

Comment:
If I remove `restrict` here, then building programs with `oshc++` still fails because of the `restrict` from the definitions in `shmemx_c_func.h4.in`.  Is there a way to work around this without removing `restrict` from the C API?

Reply:
We could do something like `#define SHMEM_QUALIFIER_RESTRICT` to `restrict` or nothing depending on the translation environment, so the qualifier is scraped off of the C bindings when using a C++ compiler.  In practice, I would expect this to work, although I'm not sure whether having the definition use restrict and the declaration not causes undefined behavior in the C language spec.

Reply:
Honestly, the best solution is probably not to use restrict.  🤕